src/main/java/org/opensearch/flowframework/workflow/RegisterAgentStep.java

```diff
@@ -101,84 +101,55 @@ public void onFailure(Exception e) {
-                        break;
-                    case LAST_UPDATED_TIME_FIELD:
-                        lastUpdateTime = Instant.ofEpochMilli((Long) entry.getValue());
-                        break;
-                    case APP_TYPE_FIELD:
-                        appType = (String) entry.getValue();
-                        break;
-                    default:
-                        break;
-                }
-            }
-        }
+        Set<String> requiredKeys = Set.of(NAME_FIELD, TYPE);
+        Set<String> optionalKeys = Set.of(
+            DESCRIPTION_FIELD,
+            LLM_MODEL_ID,
+            LLM_PARAMETERS,
+            TOOLS_FIELD,
+            PARAMETERS_FIELD,
+            MEMORY_FIELD,
+            CREATED_TIME,
+            LAST_UPDATED_TIME_FIELD,
+            APP_TYPE_FIELD
+        );
+
+        try {
+            Map<String, Object> inputs = ParseUtils.getInputsFromPreviousSteps(
+                requiredKeys,
+                optionalKeys,
+                currentNodeInputs,
+                outputs,
+                previousNodeInputs
+            );
 
-        // Case when modelId is present in previous node inputs
-        if (llmModelId == null) {
-            llmModelId = getLlmModelId(previousNodeInputs, outputs);
-        }
+            String type = (String) inputs.get(TYPE);
+            String name = (String) inputs.get(NAME_FIELD);
+            String description = (String) inputs.get(DESCRIPTION_FIELD);
+            String llmModelId = (String) inputs.get(LLM_MODEL_ID);
+            Map<String, String> llmParameters = getStringToStringMap(inputs.get(PARAMETERS_FIELD), LLM_PARAMETERS);
+            List<MLToolSpec> tools = getTools(previousNodeInputs, outputs);
+            Map<String, String> parameters = getStringToStringMap(inputs.get(PARAMETERS_FIELD), PARAMETERS_FIELD);
+            MLMemorySpec memory = getMLMemorySpec(inputs.get(MEMORY_FIELD));
+            Instant createdTime = Instant.ofEpochMilli((Long) inputs.get(CREATED_TIME));
+            Instant lastUpdateTime = Instant.ofEpochMilli((Long) inputs.get(LAST_UPDATED_TIME_FIELD));
+            String appType = (String) inputs.get(APP_TYPE_FIELD);
+
+            // Case when modelId is present in previous node inputs
+            if (llmModelId == null) {
+                llmModelId = getLlmModelId(previousNodeInputs, outputs);
+            }
 
-        // Case when modelId is not present at all
-        if (llmModelId == null) {
-            registerAgentModelFuture.completeExceptionally(
-                new FlowFrameworkException("llm model id is not provided", RestStatus.BAD_REQUEST)
-            );
-            return registerAgentModelFuture;
-        }
+            // Case when modelId is not present at all
+            if (llmModelId == null) {
+                registerAgentModelFuture.completeExceptionally(
+                    new FlowFrameworkException("llm model id is not provided", RestStatus.BAD_REQUEST)
```

Comment:
Please add workflowID and step (node) ID to this error message. (Can handle in a future PR)